packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/common/documentUtils.ts

```diff
@@ -20,17 +20,20 @@ export function getDocFieldValue(
   index?: number,
   span?: Location | TextSpan
 ): string | undefined {
-  let fieldText: string | undefined;
+  let fieldText: string | object | undefined;
 
   const documentFieldArray = document[field];
   if (!Array.isArray(documentFieldArray) && !index) {
     fieldText = documentFieldArray;
   } else {
     fieldText = documentFieldArray?.[index ?? 0];
   }
-  if (!fieldText || !span) {
+  if (typeof fieldText?.['table_text'] === 'string') {
```

Comment:
Why this change?

Reply:
Highlighting went wrong when there is `table` field in a document because it contains an object instead of text. Also, there is text_mappings for between bboxes and text in `table` fields. That code is to specially handles the `table` field to get field text referred in the text_mappings.

As the code is not clear, I updated the code and added a comment.